globus_cli/services/transfer/share/create.py

```diff
@@ -9,15 +10,14 @@
 @click.command('create', help='Create a new Share, hosted on an Endpoint')
 @common_options
 @endpoint_create_and_update_opts(create=True, shared_ep=True)
-@click.option('--host-endpoint-id', required=True,
-              help='ID of the endpoint on which this Share is hosted')
-def share_create(host_endpoint_id, display_name, description, organization,
+@endpoint_id_arg(metavar='HOST_ENDPOINT_ID')
+def share_create(endpoint_id, display_name, description, organization,
                  contact_email, contact_info, info_link, public):
```

Comment:
I noticed that issue with shares while doing this work, but decided to fix it later and just replicate the current behavior with the new parsing tools.
It will probably take an `ENDPOINT_PLUS_REQPATH`.

it seems pretty reasonable for `assemble_generic_doc()` -- whose sole purpose in this world is to help build Transfer document bodies -- would be able to apply the "safe stringify" to UUIDs at least.
In case of nested docs, it shouldn't try to do too much and risk accidentally storing a `str(subdict)` in the resulting JSON body.